docs/entity/components.md

```diff
@@ -526,6 +526,54 @@ See the following example:
 }
 ```
 
+## `Index`
+
+A field used to specify [index](https://docs.splunk.com/Documentation/Splunk/latest/Admin/Inputsconf#Scripted_Input:).
+
+<h3> Attributes </h3>
+
+| Property     | Type    | Description                                                                                                                      |
+|--------------|---------|----------------------------------------------------------------------------------------------------------------------------------|
+| type*        | string  | To specify the type of entity to be rendered in inputs or configuration form.                                                    |
+| field*       | string  | To define a particular entity field.                                                                                             |
+| label*       | string  | Text displayed next to entity field.                                                                                             |
+| defaultValue | string  | The initial input value.                                                                                                         |
+| help         | string  | Help text gives context about a fields input, such as how the input will be used. It is displayed directly below an input field. |
+| required     | string  | To specify whether the field is required or not.                                                                                 |
+
+
+See the following examples:
+
+Minimal definition:
+
+```json
+{
+    "type": "index",
+    "field": "index",
+    "label": "Index"
+}
+```
+
+Full definition:
+
+```json
+{
+    "type": "index",
+    "field": "index",
+    "label": "Index",
+    "help": "An index is a type of data repository. Select the index in which you want to collect the events.",
+    "defaultValue": "main",
+    "required": true
+}
+```
+
+Index field has two internal validators:
```

Comment:
Should we call out `validators` and `options` in the full definition as well?

Reply:
I think adding a link to the line of code where full definition is stored would be sufficient.